src/ensembl/tests/fixtures/entity-viewer/product.ts

```diff
@@ -1,20 +1,51 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 import faker from 'faker';
 import times from 'lodash/times';
 
 import {
-  ProteinDomainsResources,
+  ProteinDomain,
   Product,
   ProductType
 } from 'src/content/app/entity-viewer/types/product';
 
-export const createProduct = (): Product => {
-  const length = faker.random.number({ min: 10, max: 100 });
-  const numberOfDomains = faker.random.number({ min: 1, max: 10 });
-  const maxDomainLength = Math.floor(length / numberOfDomains);
+export const createProduct = (fragment: Partial<Product> = {}): Product => {
+  const length = fragment?.length || faker.random.number({ min: 10, max: 100 });
```

Comment:
Do we need to keep `fragment?.length` here since we use `...fragment` on line 40?

Reply:
Technically, we don't; but it makes the intention more obvious. Without it, it would be easy to forget that the `length` property always set on this line to `faker.random.number({ min: 10, max: 100 })` can get overwritten with a completely different length value coming from the `fragment`. I tried to remove the `fragment?.length` from this line and to re-read the body of the function, and didn't like the result.

Reply:
That's fair enough. When I saw it first, I did give me an impressing that it can be overridden.